app/endpoints/bucketlist.py

```diff
@@ -0,0 +1,124 @@
+from flask_restful import Resource, reqparse, abort
+from app.models import BucketList, BucketlistItems
+from app import db
+from app.auth import multiple_auth
+from flask import g, request
+from flask import jsonify, json
+
+
+class BucketListRepo(Resource):
+
+    decorators = [multiple_auth.login_required]
+
+    def post(self):
+        #get the the bucketlist items from the post request
+        parser = reqparse.RequestParser()
+        parser.add_argument('name', type=str,
+                            help='Name of bucket list is Required',
+                            required=True)
+        args = parser.parse_args()
+
+        bucketlist_name = args["name"]
+
+        if BucketList.query.filter_by(name=bucketlist_name).first() is not None:
+             abort(403, msg='Sorry A similar bucket list with the same name Exists!')
+
+        #create a bucketlist instance with the post data
+        bucketlist = BucketList(name=bucketlist_name, user_id=g.user.id,
+                                created_by=g.user.username)
+
+        db.session.add(bucketlist)
+        db.session.commit()
+
+        return "Bucket list created succefully with the ID: %s"% bucketlist.id, 201
+
+    def get(self, id=None):
+        searchword = request.args.get('q') or None
+        limit = request.args.get('limit') or 20
+        kwargs = {"user_id": g.user.id}
+        if searchword:
+            kwargs.update({"name": searchword})
+        all_bucket_lists = []
+        bucketlists_retrieved = BucketList.query.filter_by(**kwargs).limit(limit).all()
```

Comment:
isn't this supposed to make room for what matches the search e.g name of bucketlist with the substring..